tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_pgm_dispatch.cpp

```diff
@@ -218,17 +251,18 @@ int main(int argc, char **argv) {
     try {
         int device_id = 0;
         tt_metal::Device* device;
+        DispatchCoreType dispatch_core_type = dispatch_from_eth_g ? DispatchCoreType::ETH : DispatchCoreType::WORKER;
         if (use_trace_g) {
-            device = tt_metal::CreateDevice(device_id, 1, DEFAULT_L1_SMALL_SIZE, 900000000);
+            device = tt_metal::CreateDevice(device_id, 1, DEFAULT_L1_SMALL_SIZE, 900000000, dispatch_core_type);
         } else {
-            device = tt_metal::CreateDevice(device_id);
+            device = tt_metal::CreateDevice(device_id, 1, DEFAULT_L1_SMALL_SIZE, 900000000, dispatch_core_type);
```

Comment:
In this `if-else` block, `device` is being initialized the same way in both cases. Should it be different in each case?

Reply:
good catch, these code paths converged, will fix